support-frontend/app/controllers/StripeController.scala

```diff
@@ -52,6 +53,8 @@ class StripeController(
 
     val v2RecaptchaToken = request.body.token
 
+    val v2RecaptchaSecretKey = recaptchaConfigProvider.get().v2SecretKey
```

Comment:
Don't we need to use the correct key for a test user here?

Reply:
It doesn't seem so. I tested combinations of the special public key with random secret keys - the special public key seems to be enough.

That being said, the special secret key is still in AWS Parameter Store and is accessible through `recaptchaConfigProvider.get(isTestUser=true).v2SecretKey` if needed later on.

Reply:
have you tried it with prod credentials though? Because aren't UAT and DEV using the same keys, so it would work even if there was a mismatch between the environments.

Reply:
This is how I've set up the keys:

On stage DEV:
* touchpoint env UAT - special keys
* touchpoint env SANDBOX - special keys

On stage CODE:
* touchpoint env UAT - special keys
* touchpoint env SANDBOX - prod keys

On stage PROD:
* touchpoint env UAT - special keys
* touchpoint env PROD - prod keys

I've manually tested touchpoint environments UAT and SANDBOX in stages DEV and CODE. In stage DEV, both test users and normal users get a special No CAPTCHA for testing purposes (see screenshot added to the main description). In stage CODE, test users get the special No CAPTCHA, whereas normal users get a normal No CAPTCHA reCAPTCHA.

Does that make sense? (I might be missing something, just wanted to clarify which tests I've carried out.)

Reply:
> whereas normal users get a normal No CAPTCHA reCAPTCHA.

what is "normal No CAPTCHA reCAPTCHA"? is that the one where you just tick the box rather than selecting all the photos contained a horse or whatever?

Reply:
Yes. Normal ie not special, No CAPTCHA ie checkbox, reCAPTCHA ie images (if checkbox test is insufficient). Just added a mini-explainer at the bottom of the description

Reply:
I think line 66 is skipping the validation,need to remove the ` && !testPublicKeys(request.body.stripePublicKey)` from that line

Reply:
Ooh, great catch! I just tried it and there is indeed a problem when the correct special secret key is not used. Fixing.